src/AutoRest.CSharp/LowLevel/Generation/LowLevelClientWriter.cs

```diff
@@ -283,9 +290,23 @@ private void WriteEmptyConstructor (CodeWriter writer, LowLevelRestClient client
             writer.Line();
         }
 
-        private void WriteConstructor (CodeWriter writer, LowLevelRestClient client, bool keyCredential, BuildContext context)
+        private CSharpType? GetCredentialType (CredentialKind credentialKind)
         {
-            var ctorParams = client.GetConstructorParameters(keyCredential ? typeof(AzureKeyCredential) : typeof(TokenCredential));
+            switch (credentialKind)
+            {
+                case CredentialKind.Token:
+                    return typeof(TokenCredential);
+                case CredentialKind.Key:
+                    return typeof(AzureKeyCredential);
+                case CredentialKind.None:
+                default:
+                    return null;
+            }
+        }
+
+        private void WriteConstructor (CodeWriter writer, LowLevelRestClient client, CredentialKind credentialKind, BuildContext context)
```

Comment:
Could the `CredentialKind` enum be eliminated in favor of just the `Type` itself?

Reply:
Yeah it could be I guess. I kinda like the more explicit nature of a named enum. Thoughts Pavel?

Reply:
No strong opinion here.